controllers/helmchart_controller.go

```diff
@@ -474,6 +488,23 @@ func (r *HelmChartReconciler) buildFromHelmRepository(ctx context.Context, obj *
 		return sreconcile.ResultEmpty, err
 	}
 
+	defer func() {
+		// Cache the index if it was successfully retrieved
+		// and the chart was successfully built
+		if r.Cache != nil && chartRepo.Index != nil {
+			err := r.Cache.Set(r.Storage.LocalPath(*repo.GetArtifact()), chartRepo.Index, r.TTL)
```

Comment:
@souleb we need to ensure that cache keys will always be safe in multi-tenancy environments, as otherwise it could be used as a vector to bypass the helm repository's authentication. 

This implementation does not have that problem as local path would be in the format `/<helm-repository-name>/<chart-name>/<filename>`.
But it would be good to add a small comment here to that effect, so future changes in the cache keys format do not change that.

Reply:
Thanks @pjbgf i will add what you wrote as comment